worldedit-core/src/main/java/com/sk89q/worldedit/function/operation/ForwardExtentCopy.java

```diff
@@ -405,16 +422,53 @@ public Operation resume(RunContext run) throws WorldEditException {
             blockCopy = new RegionVisitor(region, copy, preloader);
         }
 
-        List<? extends Entity> entities;
+        Collection<Entity> entities;
         if (copyingEntities) {
-            // filter players since they can't be copied
-            entities = source.getEntities(region);
-            entities.removeIf(entity -> {
-                EntityProperties properties = entity.getFacet(EntityProperties.class);
-                return properties != null && !properties.isPasteable();
-            });
+            IQueueExtent<IQueueChunk> queue;
+            Extent ext = source instanceof AbstractDelegateExtent ex ? ex.getExtent() : source;
+            ParallelQueueExtent parallel = new ExtentTraverser<>(source).findAndGet(ParallelQueueExtent.class);
```

Comment:
This doesn't seem to work, I'm getting a `Could not find IQueueExtent instance for entity retrieval, falling back to default method.` From a quick look, `findAndGet` calls `find(Object)` instead of `find(Class)`. Replacing the call with `find((Class<Extent>) clazz)` makes that work, but it seems like I'm running in some kind of deadlock then.

Reply:
Are you able to find what causes the deadlock?

Reply:
I got the following exception after >10 seconds:
```
java.lang.NullPointerException: Cannot read field "r" because "levelChunk" is null
	at com.sk89q.worldedit.bukkit.adapter.impl.fawe.v1_20_R3.PaperweightGetBlocks_Copy.<init>(PaperweightGetBlocks_Copy.java:54) ~[FastAsyncWorldEdit-Bukkit-2.9.2-SNAPSHOT.jar:?]
	at com.sk89q.worldedit.bukkit.adapter.impl.fawe.v1_20_R3.PaperweightGetBlocks.call(PaperweightGetBlocks.java:380) ~[FastAsyncWorldEdit-Bukkit-2.9.2-SNAPSHOT.jar:?]
	at com.fastasyncworldedit.core.queue.implementation.chunk.ChunkHolder.call(ChunkHolder.java:1044) ~[FastAsyncWorldEdit-Bukkit-2.9.2-SNAPSHOT.jar:?]
	at com.fastasyncworldedit.core.queue.implementation.chunk.ChunkHolder.call(ChunkHolder.java:1015) ~[FastAsyncWorldEdit-Bukkit-2.9.2-SNAPSHOT.jar:?]
	at com.fastasyncworldedit.core.queue.implementation.chunk.ChunkHolder.call(ChunkHolder.java:35) ~[FastAsyncWorldEdit-Bukkit-2.9.2-SNAPSHOT.jar:?]
```
I'll try to find out more

Reply:
Okay that seems to happen after pasting only, not sure what went wrong with copying before, I cannot reproduce it anymore